sprinter/environment.py

```diff
@@ -573,9 +569,18 @@ def _specialize(self, reconfigure=False):
     def _copy_source_to_target(self):
         """ copy source user configuration to target """
         if self.source and self.target:
-            for k, v in self.source.items('config'):
-                # always have source override target.
-                self.target.set_input(k, v)
+            # self.target.inputs.copy_values(self.source.inputs)
+            for k in self.target.inputs.keys():
+                if self.source.has_option('config', k):
+                    cur_val = self.source.get('config', k)
+                    cur_default = self.source.inputs.get_input_default(k)
+                    new_default = self.target.inputs.get_input_default(k)
+                    # don't copy the value if it is the default and the default has changed
+                    # causes a reprompt on update when defaults change
+                    if new_default == cur_default:
+                        self.target.set_input(k, cur_val)
+                    elif cur_val != cur_default:
+                        self.target.set_input_prev(k, cur_val)
```

Comment:
This will now cause a re-prompt when updating, when the default value for an input changes.